ext/TensorKitChainRulesCoreExt/factorizations.jl

```diff
@@ -57,6 +57,7 @@ function ChainRulesCore.rrule(::typeof(TensorKit.eig!), t::AbstractTensorMap; kw
     D, V = eig(t; kwargs...)
 
     function eig!_pullback((ΔD, ΔV))
+        ΔD, ΔV = unthunk(ΔD), unthunk(ΔV)
```

Comment:
I am not sure if this is causing type ambiguities in this way, or whether the compiler is smart enough to figure this out. A simple explicit resolution is to write
```julia
function eig!_pullback((_ΔD, _ΔV))
         ΔD, ΔV = unthunk(_ΔD), unthunk(_ΔV)
```